Dockerfile

```diff
@@ -1,31 +1,47 @@
+# Stage 1 build
 FROM golang:1.16.1-buster AS builder
+RUN apt-get update && apt-get install ca-certificates -y
+ENV CGO_ENABLED=0
+
+ENV USER=bubbly
+ENV UID=10001
+RUN adduser \
+    --disabled-password \
+    --gecos "" \
+    --home "/nonexistent" \
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid "${UID}" \
+    "${USER}"
 
-WORKDIR /src
+WORKDIR $GOPATH/src/main/bubbly/
 
 COPY go.sum .
 COPY go.mod .
 
 RUN go mod download
-
-RUN apt-get update && apt-get install ca-certificates -y
+RUN go mod verify
 
 COPY . .
 
 RUN go get -u github.com/swaggo/swag/cmd/swag
 RUN swag init
 
-# compile the go binary
-RUN go build -o /bubbly
+RUN GOOS=linux GOARCH=amd64 go build -ldflags="-w -s" -o /go/bin/bubbly
 
-# target for running in cluster
-FROM debian:buster-slim
-COPY --from=builder /bubbly /bubbly
+# step 2
+FROM scratch
+# Import the user and group files from the builder.
+COPY --from=builder /etc/passwd /etc/passwd
+COPY --from=builder /etc/group /etc/group
```

Comment:
This does not look right 😮 There will be all sorts of service account information there, which may change from release to release. There got to be a better way to do this.

Now I see why you cannot run `useradd` here, it's a `scratch` image, but perhaps the easiest solution is just to append a line to each `/etc/passwd` and `/etc/group`, like Istio does [here](https://github.com/istio/istio/blob/master/docker/Dockerfile.distroless)?

I'm not sure how to best approach this. Moar research is needed 🙃

Reply:
Looks like distroless contains a user called "nonroot", this completely removes the need to copy anything 😄